api/environments/identities/views.py

```diff
@@ -254,12 +257,22 @@ def post(self, request):
             },
         )
 
+    def _get_additional_filters(self) -> Q | None:
+        if self.request.originated_from is RequestOrigin.CLIENT:
```

Comment:
We should perhaps be defensive here and use `getattr(self.request, "originated_from", None)` ?

Reply:
I'd actually prefer this part broken if we switch/modify the authentication class for some reason. Unit tests cover this so we should be safe.

Reply:
Yep, ok, sounds good 👍